GliaWidgetsTests/Sources/Coordinators/EngagementCoordinator/EngagementCoordinatorCallTests.swift

```diff
@@ -0,0 +1,136 @@
+
+
+        XCTAssertNotNil(viewController)
+        XCTAssertEqual(coordinator.interactor.state, .enqueueing(.video))
+    }
+
+    // Delegate
+
+    func test_callDelegateBack() throws {
+        coordinator = createCoordinator(withKind: .videoCall)
+        var calledEvents: [EngagementCoordinator.DelegateEvent] = []
+
+        coordinator.delegate = { event in
+            calledEvents.append(event)
+        }
+
+        coordinator.start()
+
+        let callCoordinator = coordinator.coordinators.first {$0 is CallCoordinator} as? CallCoordinator
+        callCoordinator?.delegate?(.back)
+
+        callAfterTimeout {
+            XCTAssertTrue(calledEvents.contains(.minimized))
+        }
+    }
+
+    func test_callDelegateBackCallFromChat() throws {
+        var calledEvents: [EngagementCoordinator.DelegateEvent] = []
+
+        coordinator.delegate = { event in
+            calledEvents.append(event)
+        }
+
+        coordinator.start()
+
+        let chatCoordinator = coordinator.coordinators.first { $0 is ChatCoordinator } as? ChatCoordinator
+
+        let mediaUpgradeOffer = try! CoreSdkClient.MediaUpgradeOffer(type: .audio, direction: .twoWay)
+        chatCoordinator?.delegate?(
+            .mediaUpgradeAccepted(
+                offer: mediaUpgradeOffer,
+                answer: { answer, success in success?(true, nil) }
+            )
+        )
+
+        let callCoordinator = coordinator.coordinators.first {$0 is CallCoordinator} as? CallCoordinator
+        callCoordinator?.delegate?(.back)
+
+        callAfterTimeout {
+            let topmostViewController = coordinator.navigationPresenter.viewControllers.first as? ChatViewController
+
+            XCTAssertNotNil(topmostViewController)
+        }
+    }
+
+    func test_callDelegateEngaged() throws {
+        coordinator = createCoordinator(withKind: .videoCall)
+        
+        coordinator.start()
+
+        try showGliaViewController()
+        let callCoordinator = coordinator.coordinators.first { $0 is CallCoordinator } as! CallCoordinator
+
+        callCoordinator.delegate?(.engaged(operatorImageUrl: URL.mock.absoluteString))
+
+        switch coordinator.gliaViewController!.bubbleKind {
+        case .userImage(let url):
+            XCTAssertEqual(url, URL.mock.absoluteString)
+        default: XCTFail()
+        }
+    }
+
+    func test_callDelegateMinimize() throws {
+        coordinator = createCoordinator(withKind: .videoCall)
+        var calledEvents: [EngagementCoordinator.DelegateEvent] = []
+
+        coordinator.delegate = { event in
+            calledEvents.append(event)
+        }
+
+        coordinator.start()
+
+        let callCoordinator = coordinator.coordinators.first {$0 is CallCoordinator} as? CallCoordinator
+        callCoordinator?.delegate?(.minimize)
+
+        callAfterTimeout {
+            XCTAssertEqual(calledEvents.last, .minimized)
+        }
+    }
+
+//    func test_callDelegateFinished() throws {
```

Comment:
will we use this test in future?

Reply:
Nope, fixed bbb78205f0b4d4a0c13507fd9f24409ed42d3523